### ISSUE TITLE
[feature] make system_requirements error logger more informative

### ISSUE THREAD

**Issue by AndreyMlashkin** [conan-io/conan#11706]
currently, error log looks like:
```
xorg/system: ERROR: while executing system_requirements(): Can't update because tools.system.package_manager:mode is 'check'.Please update packages manually or set 'tools.system.package_manager:mode' to 'install' in the [conf] section of the profile, or in the command line using '-c tools.system.package_manager:mode=install'
ERROR: Error in system requirements
```
If should contain a list, which packages are missing.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Can you please report which tool are you using inside ``system_requirements()``? Because there are 2, the legacy and the new one in ``from conan.tools``.

**Comment by AndreyMlashkin** [comment]
I am not sure, what do you exactly mean.
I am installing official xorg/system
https://github.com/conan-io/conan-center-index/blob/master/recipes/xorg/all/conanfile.py

**Comment by memsharded** [comment]
Good, that is using the new tooling in ``from conan.tools.system.package_manager import Apt, Dnf, PacMan, Pkg, Yum, Zypper``

**Comment by czoido** [comment]
Hi @AndreyMlashkin,
Thanks for the feedback. I can open a PR to improve the message. What information do you miss in the message? Maybe it can also be a problem of lack of visibility in the Conan documentation? The new package_manager tools are documented [here](https://docs.conan.io/en/latest/reference/conanfile/tools/system/package_manager.html).

**Comment by AndreyMlashkin** [comment]
I would like to see, which packages are missing.
It would be perfect to have the full list of missing dependencies

**Comment by czoido** [comment]
> I would like to see, which packages are missing. It would be perfect to have the full list of missing dependencies

I think the problem is that the recipe in conan-center-index is forcing an update for the packages but that will make that something like `apt-get update` is executed and that is not allowed if the  configuration `package_manager:mode` is different from `install`. Let me mark this as look into so I can discuss the implications of this with the team.

**Comment by czoido** [comment]
@AndreyMlashkin

Did you try to set the conf: `tools.system.package_manager:mode=install`?

You can do that in the command line like:

```
conan create ... -c tools.system.package_manager:mode=install
```
Or in the `[conf]` section of your [Conan profile](https://docs.conan.io/en/latest/reference/config_files/global_conf.html#configuration-in-your-profiles).

**Comment by GavinNL** [comment]
My CI is failing with the same error
```
xorg/system: ERROR: while executing system_requirements(): Can't update because tools.system.package_manager:mode is 'check'.Please update packages manually or set 'tools.system.package_manager:mode' to 'install' in the [conf] section of the profile, or in the command line using '-c tools.system.package_manager:mode=install'
```

My CI has sudo, but not sure what to do about this. I tried adding the following but it didn't work
```
conan install .. -c tools.system.package_manager:mode=install
```

Is there anything else I'm supposed to add?

**Comment by czoido** [comment]
Hi @GavinNL,
If you also add `-c tools.system.package_manager:sudo=True` the package manager will be invoked with sudo:

```
conan install .. -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=True
```

Hope this helps.

**Comment by GavinNL** [comment]
Ahh okay. looks like that worked. Thanks!

Is there a way to get the list of system packages required by the conanfile? something like:

```
conan list-required-system-packages ../conanfile.txt
```

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin
> 
> Did you try to set the conf: `tools.system.package_manager:mode=install`?
> 
> You can do that in the command line like:
> 
> ```
> conan create ... -c tools.system.package_manager:mode=install
> ```
> 
> Or in the `[conf]` section of your [Conan profile](https://docs.conan.io/en/latest/reference/config_files/global_conf.html#configuration-in-your-profiles).

I don't need to install the dependencies, I only want to print missing ones. The same as a call
`conan install my_dependency/1.1.1@` would do if you don't specify --build missing

**Comment by czoido** [comment]
> Ahh okay. looks like that worked. Thanks!
> 
> Is there a way to get the list of system packages required by the conanfile? something like:
> 
> ```
> conan list-required-system-packages ../conanfile.txt
> ```

@GavinNL we don't have a dedicated command for that but you could iterate the dependencies in the conanfile's `generate()` method and get the list of system libs. Something like:

```python
def generate(self):
    system_libs = []
    for dep in self.dependencies.values():
        if dep.cpp_info.has_components:
            sorted_components = dep.cpp_info.get_sorted_components().items()
            for comp_name, comp_cpp_info in sorted_components:
                system_libs.extend(comp_cpp_info.system_libs)
        else:
            system_libs.extend(dep.cpp_info.system_libs)

    print(system_libs)
```

In Conan 2.0, there's the possibility to add your own custom commands, so you could make your own command for that. You can read about that [in the Conan 2.0 documentation](https://docs.conan.io/en/2.0/reference/commands/custom_commands.html#reference-commands-custom-commands).

**Comment by czoido** [comment]
@AndreyMlashkin 
We have merged https://github.com/conan-io/conan/pull/11712 and will be released in 1.51, from that version you will see what packages are missing being in mode=check, so this should solve your issue.
